ion/bitstream.go

```diff
@@ -541,41 +541,55 @@ func (b *bitstream) ReadTimestamp() (time.Time, error) {
 		}
 	}
 
-	nsecs, err := b.readNsecs(len)
+	nsecs, overflow, err := b.readNsecs(len)
 	if err != nil {
 		return time.Time{}, err
 	}
 
 	b.state = b.stateAfterValue()
 	b.clear()
 
-	return tryCreateTimeWithNSecAndOffset(ts, nsecs, offset)
+	return tryCreateTimeWithNSecAndOffset(ts, nsecs, overflow, offset)
 }
 
-func tryCreateTimeWithNSecAndOffset(ts []int, nsecs int, offset int64) (time.Time, error) {
+func tryCreateTimeWithNSecAndOffset(ts []int, nsecs int, overflow bool, offset int64) (time.Time, error) {
 	date := time.Date(ts[0], time.Month(ts[1]), ts[2], ts[3], ts[4], ts[5], nsecs, time.UTC)
 	// time.Date converts 2000-01-32 input to 2000-02-01
 	if ts[0] != date.Year() || time.Month(ts[1]) != date.Month() || ts[2] != date.Day() {
 		return time.Time{}, fmt.Errorf("ion: invalid timestamp")
 	}
 
+	if overflow {
+		date = date.Add(time.Second)
+	}
 	return date.In(time.FixedZone("fixed", int(offset)*60)), nil
 }
 
-// ReadNsecs reads the fraction part of a timestamp and truncates it to nanoseconds.
-func (b *bitstream) readNsecs(len uint64) (int, error) {
+// ReadNsecs reads the fraction part of a timestamp and rounds to nanoseconds.
```

Comment:
Please add a sentence to this comment explaining the new `(int, bool, error)` return type.

Reply:
Updated.